scio-core/src/main/scala/com/spotify/scio/util/ScioUtil.scala

```diff
@@ -82,5 +85,41 @@ private[scio] object ScioUtil {
       s"Could not get dataflow metrics of ${getMetrics.getJobId} in ${getMetrics.getProjectId}")
   }
 
+  def tempLocation(options: PipelineOptions): String = {
+    Option(Try(options.as(classOf[GcpOptions]))
```

Comment:
Not a huge fan of such complex chain. IMO the original `if/else` is more readable.